src/components/gallery/Gallery.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useCallback, useState} from 'react';
+import {
+  Image,
+  ImageSourcePropType,
+  StyleSheet,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import {Indicator} from 'components/gallery/Indicator';
+
+interface GalleryProps {
+  images: ImageSourcePropType[];
+}
+
+export const Gallery: React.FC<GalleryProps> = ({images = []}) => {
+  const [activeImageIdx, setActiveImageIdx] = useState(0);
+
+  const selectImage = useCallback((index: number) => {
+    setActiveImageIdx(index);
+  }, []);
+
+  const showPrevImage = useCallback(() => {
+    selectImage(activeImageIdx === 0 ? images.length - 1 : activeImageIdx - 1);
+  }, [activeImageIdx, images.length, selectImage]);
+
+  const showNextImage = useCallback(() => {
+    selectImage(activeImageIdx === images.length - 1 ? 0 : activeImageIdx + 1);
+  }, [activeImageIdx, images.length, selectImage]);
```

Comment:
You don't pass these functions to other components, so there is no need to use `useCallback` here

Reply:
But I pass them as `onPress` property of w instances of `TouchableOpacity` below. Isn't callback recommended in this case?

Reply:
@daniil-gorbunov not really. Of course you can use `useCallback`, but the effort will be minimal here.